### PR TITLE
Update changelogger validation to only affect PRs against trunk

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the changelogger validation to only affect PRs against trunk. The motivation for this change is that it allows for PRs against a feature-branch without requiring a changelog entry, or PRs against a release branch (which often build the changelog) without requiring a changelog entry.

### How to test the changes in this Pull Request:

1. Fork repo.
2. Enable actions in your fork.
3. Merge this PR to `trunk` in your fork.
4. Create a PR against a branch other than `trunk` without creating a changelog entry. Verify that the action isn't run.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33214/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Fix syntax error in changelog workflow` ff0a3ad9baf86d718bfa0c2db91e4e241e78e8ac
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: .github/workflows/pr-lint-monorepo.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 name: Run lint checks potentially affecting projects across the monorepo
 on: pull_request
+  branches:
+    - trunk
```
Comment (psealock) [inline .github/workflows/pr-lint-monorepo.yml]: I think this [syntax](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) isn't quite right. I'm thinking its more like this
```
on:
  pull_request:
    branches:    
      - 'trunk'
```

**Review by psealock — COMMENTED** [review]
Nice, this is a good idea particularly for feature branches.

The check didn't run on my [test PR](https://github.com/woocommerce/woocommerce/pull/33222), but I noticed it hasn't run on this PR either.

File: .github/workflows/pr-lint-monorepo.yml (hunk 1 above)
Comment (jonathansadowski) [reply .github/workflows/pr-lint-monorepo.yml]: Good catch @psealock, I didn't notice that it hadn't been running on this PR. This should be ready for another look.

**Comment by jonathansadowski** [comment]
Thanks for the review! @psealock would you be able to give this another look?

**Review by psealock — APPROVED** [review]
Looks good to me, working as expected. 🚢

**Pull request merged**